web/frontend/src/layout/NavBar.tsx

```diff
@@ -55,248 +51,202 @@ const NavBar: FC = () => {
+                    </div>
+                  </div>
 
-              <svg
-                className={`${menuToggle ? 'block' : 'hidden'} h-6 w-6`}
-                xmlns="http://www.w3.org/2000/svg"
-                fill="none"
-                viewBox="0 0 24 24"
-                stroke="currentColor"
-                aria-hidden="true">
-                <path
-                  strokeLinecap="round"
-                  strokeLinejoin="round"
-                  strokeWidth="2"
-                  d="M6 18L18 6M6 6l12 12"
-                />
-              </svg>
-            </button>
-          </div>
+                  <div className="py-6 px-5 space-y-6">
+                    <div className="grid grid-cols-2 gap-y-4 gap-x-8">
+                      {availableLanguages.map(
+                        (lang) =>
+                          i18n.language !== lang && (
+                            <Popover.Button key={lang}>
+                              <div
+                                className="text-base font-medium text-gray-900 hover:text-gray-700"
+                                onClick={() => i18n.changeLanguage(lang)}>
+                                {t(lang)}
+                              </div>
+                            </Popover.Button>
+                          )
+                      )}
+                    </div>
+                  </div>
 
-          <div className="flex-1 flex items-center justify-center sm:justify-start">
+                  <div className="py-6 px-5 space-y-6">
+                    {authCtx.isLogged && (
+                      <div>
+                        Logged as {authCtx.firstname} {authCtx.lastname}
+                      </div>
+                    )}
+                    <div>
+                      {authCtx.isLogged ? (
+                        <div onClick={handleLogout}>
+                          <Popover.Button className="w-full flex items-center justify-center px-4 py-2 border  rounded-md shadow-sm text-base font-medium">
+                            <LogoutIcon className="-ml-1 mr-2 h-5 w-5" aria-hidden="true" />
+                            {t('logout')}
+                          </Popover.Button>
+                        </div>
+                      ) : (
+                        <div onClick={() => handleLogin(loginError, setLoginError)}>
+                          <Popover.Button className="w-full flex items-center justify-center px-4 py-2 border border-transparent rounded-md shadow-sm text-base font-medium text-white bg-indigo-600 hover:bg-indigo-700">
+                            <LoginIcon className="-ml-1 mr-2 h-5 w-5" aria-hidden="true" />
+                            {t('login')}
+                          </Popover.Button>
+                        </div>
+                      )}
+                    </div>
+                  </div>
+                </div>
+              </Popover.Panel>
+            </Transition>
+          </Popover>
+          {/* End Mobile menu */}
```

Comment:
To improve readability I would move the mobile menu to its own function component, in the same file. Same for the right/left side if this is not too complicated.